src/content/en/updates/2019/08/native-file-system.md

```diff
@@ -0,0 +1,457 @@
+user to select multiple files, or directories, or different file types.
+Without any options specified), the file picker allows the user to select a
+single file, perfect for our text editor.
+
+Like many other powerful APIs, calling `chooseFileSystemEntries()` must be
+done in a [secure context][secure-contexts], and must be called from within
+a user gesture.
+
+```js
+let fileHandle;
+butOpenFile.addEventListener('click', async (e) => {
+  fileHandle = await window.chooseFileSystemEntries();
+  // Do something with the file handle
+});
+```
+
+Once the user selects a file, `chooseFileSystemEntries()` returns a handle,
+in this case a [`FileSystemFileHandle`][fs-file-handle] that contains the
+properties and methods needed to interact with the file.
+
+It’s helpful to keep a reference to the file handle around so that it
+can be used later. It’ll be needed to save changes back to the file, or
+any other file operations. In the next few milestones, installed
+Progressive Web Apps will also be able to save the handle to IndexedDB
+and persist access to the file across page reloads.
+
+#### Read a file from the file system
+
+Now that you have a handle to a file, you can get the properties of the file,
+or access the file itself. For now, let’s simply read the contents of the
+file. Calling `handle.getFile()` returns a [`File`][file-api-spec] object,
+which contains binary data as a blob. To get the data from the blob, call
+one of the reader methods (`slice()`, `stream()`, `text()`, `arrayBuffer()`).
+
+```js
+const file = await fileHandle.getFile();
+const contents = await file.text();
+```
+
+#### Putting it all together
+
+Putting it all together, when the user clicks the Open button, the browser
+shows a file picker. Once they’ve selected a file, the app reads the
+contents and puts them into a `<textarea>`.
+
+```js
+let fileHandle;
+butOpenFile.addEventListener('click', async (e) => {
+  fileHandle = await window.chooseFileSystemEntries();
+  const file = await fileHandle.getFile();
+  const contents = await file.text();
+  textArea.value = contents;
+});
+```
+
+### Write the file to the local file system {: #write-file }
+
+In the text editor, there are two ways to save a file: Save, and Save As.
+Save simply writes the changes back to the original file using the file
+handle we got earlier. But Save As creates a new file, and thus requires a
+new file handle.
+
+#### Create a new file
+
+The `chooseFileSystemEntries()` API with `{type: 'saveFile'}` will show the
+file picker in “save” mode, allowing the user to pick a new file they want
+to use for saving. For the text editor, I also wanted it to automatically
+add a `.txt` extension, so I provided some additional parameters.
+
+```js
+function getNewFileHandle() {
+  const opts = {
+    type: 'saveFile',
+    accepts: [{
+      description: 'Text file',
+      extensions: ['txt'],
+      mimeTypes: ['text/plain'],
+    }],
+  };
+  const handle = window.chooseFileSystemEntries(opts);
```

Comment:
Isn't this an async function? `await window.chooseFileSystemEntries(opts)`
Then the `getNewFileHandle` function itself needs to be marked as `async`, too.